### PR TITLE
giflossy 1.82.1 (new formula)

### PR DESCRIPTION
**Pull request by denji**
Based on **gifsicle** which implements lossy LZW compression. It can reduce animgif file sizes by 30%—50% at a cost of some dithering/noise.

https://pornel.net/lossygif @pornel

### CODE REVIEW

**Comment by vszakats** [comment]
I suggest updating `homepage` to use SSL/TLS:
https://pornel.net/lossygif

File: Library/Formula/gifsicle.rb [hunk 1]
```diff
@@ -17,6 +17,8 @@ class Gifsicle < Formula
     depends_on "automake" => :build
   end
 
+  conflicts_with "giflossy"
```
Comment (DomT4) [inline Library/Formula/gifsicle.rb]: A `:because => "Blah" ` is useful in these situations.

File: Library/Formula/giflossy.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+class Giflossy < Formula
+  homepage "https://pornel.net/lossygif"
+  url "https://github.com/pornel/giflossy/archive/lossy/1.82.1.tar.gz"
+  sha256 "a0d048f0c2274c81532a988d2f3ea16c3f1cbb6878f13deeb425d34826e4ed23"
+  head "https://github.com/pornel/giflossy.git"
+
+  conflicts_with "gifsicle"
```
Comment (DomT4) [inline Library/Formula/giflossy.rb]: A `:because => "Blah" ` is useful in these situations.

**Comment by jacknagel** [comment]
```
==> brew style giflossy
== /usr/local/Library/Formula/giflossy.rb ==
C:  8: 17: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
```

**Comment by denji** [comment]
@jacknagel :rocket: Ready formula to push!

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

File: Library/Formula/giflossy.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Giflossy < Formula
+  homepage "https://pornel.net/lossygif"
```
Comment (mikemcquaid) [inline Library/Formula/giflossy.rb]: Please add a `desc` (description)

File: Library/Formula/giflossy.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  desc "Lossy LZW compression animated GIF reduce file sizes by 30%—50%"
```
Comment (DomT4) [inline Library/Formula/giflossy.rb]: The `desc` needs to go above the `homepage`. Should be fine otherwise.

File: Library/Formula/gifsicle.rb [hunk 5]
```diff
@@ -17,6 +17,9 @@ class Gifsicle < Formula
     depends_on "automake" => :build
   end
 
+  conflicts_with "giflossy",
+    :because => 'both install an `gifsicle` binary'
```
Comment (DomT4) [inline Library/Formula/gifsicle.rb]: Double quotes around the `both install` line.

File: Library/Formula/giflossy.rb [hunk 6]
```diff
@@ -0,0 +1,33 @@
+class Giflossy < Formula
+  desc "Lossy LZW compression animated GIF reduce file sizes by 30%—50%"
```
Comment (DomT4) [inline Library/Formula/giflossy.rb]: Can we tweak this a bit? It doesn't read entirely coherently at the moment. Maybe something like:

```ruby
desc "Lossy LZW compression, reduces GIF file sizes by 30-50%"
```

Or something?

Not sure what the policy is on commas in descriptions.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew/commit/88fe4488f9e5a5d39337e115807e0feb35289a4f. Thanks for your contribution to Homebrew, we appreciate it! :smiley_cat:

**Pull request closed without merging**